util/augmentation/audio_augmentation/augmentation.py

```diff
@@ -0,0 +1,43 @@
+import json
+import random
+from .speed_pertubation import SpeedPertubation
+
+class AugmentationPipeline(object):
+    def __init__(self, augmentation_config, random_seed=0.001):
+        self._rng = random.Random(random_seed)
+        self._augmentors, self._rates = self._parse_pipeline_from_json(augmentation_config)
+
+    def transform(self, samples, fs):
+        """
+        Run the augmentation pipeline for audio_augmentation
+        as per the config file.
+
+        :param samples:
+        :type: tensor
+        :param fs:
+        :type: tensor
+        """
+        processed_samples = samples
+        for augmentor, rate in zip(self._augmentors, self._rates):
+            if self._rng.uniform(0., 1.) < rate:
+                processed_samples = augmentor.transform(samples, fs)
```

Comment:
If there are more than one transformations in the pipeline, this will only apply the last one, as you're transforming from `samples` every time. It should transform `processed_samples` instead.